projects/angular-jwt/src/lib/jwt.interceptor.ts

```diff
@@ -98,31 +96,38 @@ export class JwtInterceptor implements HttpInterceptor {
   }
 
   handleInterception(
-    token: string | null,
+    token: string | Promise<string> | null,
     request: HttpRequest<any>,
     next: HttpHandler
-  ) {
+  ): Observable<HttpEvent<any>> {
     const authScheme = this.jwtHelper.getAuthScheme(this.authScheme, request);
-    let tokenIsExpired = false;
 
     if (!token && this.throwNoTokenError) {
       throw new Error("Could not get token from tokenGetter function.");
     }
 
-    if (this.skipWhenExpired) {
-      tokenIsExpired = token ? this.jwtHelper.isTokenExpired(token) : true;
-    }
+    return from(Promise.resolve(token)).pipe(
+      mergeMap<string, Observable<HttpEvent<any>>>((tokenValue: string) => {
+        let tokenIsExpired = false;
 
-    if (token && tokenIsExpired && this.skipWhenExpired) {
-      request = request.clone();
-    } else if (token) {
-      request = request.clone({
-        setHeaders: {
-          [this.headerName]: `${authScheme}${token}`,
-        },
-      });
-    }
-    return next.handle(request);
+        if (this.skipWhenExpired) {
+          tokenIsExpired = token
+            ? (this.jwtHelper.isTokenExpired(tokenValue) as boolean)
+            : true;
+        }
+
+        if (tokenValue && tokenIsExpired && this.skipWhenExpired) {
+          request = request.clone();
+        } else if (tokenValue) {
+          request = request.clone({
+            setHeaders: {
+              [this.headerName]: `${authScheme}${tokenValue}`,
+            },
+          });
+        }
+        return next.handle(request);
+      })
+    );
```

Comment:
Changes in this method are almost all unneeded:  
- The entrypoint of the interceptor is the `intercept` method below and it already handles the fact that the token can be a promise.
- the `isTokenExpired` method, though can (according to the signature only) return a Promise (which in a synchronous case will neven happen). The ugly fix is to cast the value as boolean as we know it can only be boolean.

Reply:
@piotaixr thanks for the review. Course, totally overlooked the promise handling in the in intercept function 🙈

Reply:
@piotaixr made changes. Tried to hit the review button and it removed the auth0/developer-relations-admin instead 🤦